src/navigator.js

```diff
@@ -0,0 +1,154 @@
+    });
+    server.on('request', function(req, res) {
+        network.then(function(network){
+            if (req.url.indexOf(path) === 0) {
+                var start = path.length;
+                var len = req.url.length;
+                var q = req.url.indexOf('?', start);
+                var qs = q > 0 ? req.url.substring(q) : '';
+                var end = q > 0 ? q : len;
+                var nodeId = decodeURI(req.url.substring(start, end));
+                if (!nodeId) {
+                    return index.then(function(index) {
+                        return listing(network, index, path, qs, res);
+                    });
+                } else {
+                    var vnid = qs && querystring.parse(qs.substring(1)).vnid;
+                    var node = _.find(network.graph.nodes, function(node){
+                        return node.id == nodeId;
+                    });
+                    var vnis = facades[nodeId] && vnid ?
+                        _.object([vnid], [facades[nodeId].vnis[vnid]]) :
+                        facades[nodeId] ? facades[nodeId].vnis : undefined;
+                    if (!qs && !vnis)
+                        return redirect('?view', res);
+                    else if (!qs)
+                        return redirect('?data', res);
+                    else if (qs.indexOf('?data') === 0)
+                        return data(vnis[vnid || ''], res);
+                    else if (qs.indexOf('?output') === 0)
+                        return output(vnis[vnid || ''], res);
+                    else if (qs.indexOf('?view') === 0)
+                        return render.then(function(render) {
+                            return view(network, facades, render, path, vnis, vnid, nodeId, qs, res);
+                        });
+                }
+            }
+        }).catch(function(error){
+            res.writeHead(500, {'Content-Type': 'text/plain'});
+            res.end(error.stack ? error.stack : error);
+        });
+    });
+};
+
+function listing(network, index, path, qs, res) {
+    res.setHeader('Content-Type', 'text/html');
+    res.end(index({
+        path: path,
+        search: qs || '?view',
+        name: network.graph.name,
+        nodes: network.graph.nodes
+    }));
+}
+
+function redirect(target, res) {
+    res.writeHead(303, {'Location': target});
+    res.end();
+}
+
+function data(vni, res) {
+    if (vni) {
+        res.writeHead(200, {'Content-Type': 'application/json'});
+        res.end(JSON.stringify(vni.outputState.data, null, 2));
+    } else {
+        res.writeHead(404);
+        res.end("No output state available");
+    }
+}
+
+function output(vni, res) {
+    if (vni) {
+        res.writeHead(200, {'Content-Type': 'application/json'});
+        res.end(JSON.stringify(vni.outputState, null, 2));
+    } else {
+        res.writeHead(404);
+        res.end("No output state available");
+    }
+}
+
+function view(network, facades, render, path, vnis, vnid, nodeId, qs, res) {
```

Comment:
Should probably put some javadoc comments on this.  We all know what these parameters are, but those out side the project may not understand the facdes, or the vnis, vnid, nodeId...